src/ray/gcs/store_client/redis_store_client.cc

```diff
@@ -448,50 +388,59 @@ void RedisStoreClient::RedisScanner::OnScanCallback(
     } else {
       cursor_ = cursor;
     }
+    // Result is an array of key-value pairs.
     RAY_CHECK(scan_result.size() % 2 == 0);
     for (size_t i = 0; i < scan_result.size(); i += 2) {
-      auto key = GetKeyFromRedisKey(
-          external_storage_namespace_, std::move(scan_result[i]), table_name_);
-      results_.emplace(std::move(key), std::move(scan_result[i + 1]));
+      results_.emplace(std::move(scan_result[i]), std::move(scan_result[i + 1]));
```

Comment:
can you tell me a little bit about why this is changed?

Reply:
Previously:
- req: `HSCAN hash_with_cluster_id MATCH Jobs*`
- resp: `Jobs@job1 job1_value`
- GetKeyFromRedisKey extracts: `job1`
- Saves to results_: `job1: job1_value`

Now:
- req: `HSCAN hash_with_cluster_id_for_Jobs`
- resp: `job1 job1_value`
- no need to call GetKeyFromRedisKey, key is directly job_id (job1)
- Saves to results_: `job1: job1_value`